src/lib/alloc.c

```diff
@@ -223,14 +258,16 @@ static void *alloc_block(struct mm_heap *heap, int level,
 
 /* allocates continuous blocks */
 static void *alloc_cont_blocks(struct mm_heap *heap, int level,
-	uint32_t caps, size_t bytes)
+	uint32_t caps, size_t bytes, uint32_t alignment)
 {
 	struct block_map *map = &heap->map[level];
 	struct block_hdr *hdr;
+	struct alignment_hdr *a_hdr;
 	void *ptr;
 	unsigned int start = map->first_free;
 	unsigned int current;
-	unsigned int count = bytes / map->block_size;
+	unsigned int count = (bytes + alignment + sizeof(*a_hdr))
```

Comment:
Don't you modify the bytes before calling `alloc_cont_blocks`?

Reply:
I feel like this line will break continuous block allocation. Why do you apply alignment to the variable counting amount of blocks? It doesn't make sense to me